code/_onclick/hud/new_player.dm

```diff
@@ -256,13 +280,56 @@
 	set_button_status(TRUE)
 	UnregisterSignal(SSticker, COMSIG_TICKER_ENTER_PREGAME)
 
-/atom/movable/screen/lobby/button/settings
+//Subtype the bottom buttons away so the collapse/expand shutter goes behind them
+/atom/movable/screen/lobby/button/bottom
+	layer = LOBBY_BOTTOM_BUTTON_LAYER
 	icon = 'icons/hud/lobby/bottom_buttons.dmi'
+	always_shown = TRUE
+
+/atom/movable/screen/lobby/button/bottom/Initialize(mapload)
+	. = ..()
+	if(hud)
+		RegisterSignal(hud, COMSIG_HUD_LOBBY_COLLAPSED, PROC_REF(collapse_button))
+		RegisterSignal(hud, COMSIG_HUD_LOBBY_EXPANDED, PROC_REF(expand_button))
+
+/atom/movable/screen/lobby/button/bottom/Destroy()
+	. = ..()
+	if(hud)
+		UnregisterSignal(hud, list(COMSIG_HUD_LOBBY_COLLAPSED, COMSIG_HUD_LOBBY_EXPANDED))
+
+///Animates moving the button off-screen, disabling it
+/atom/movable/screen/lobby/button/bottom/proc/collapse_button()
+	SIGNAL_HANDLER
+	var/matrix/transformation = matrix()
+	transform = transformation
+	//wait for the shutter to come down
+	animate(src, transform = transformation, time = SHUTTER_MOVEMENT_DURATION + SHUTTER_WAIT_DURATION)
+	//then pull the buttons up with the shutter
+	animate(transform = transformation.Translate(x = 0, y = 146), time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_IN)
+	if(istype(src, /atom/movable/screen/lobby/button/bottom/poll))
+		var/atom/movable/screen/lobby/button/bottom/poll/poll_button = src
+		if(!poll_button.new_poll) //don't deactivate the poll button unless a poll is up (it's already inactive)
+			return
+	set_button_status(FALSE)
+
+///Animates moving the button back into place, enabling it
+/atom/movable/screen/lobby/button/bottom/proc/expand_button()
+	SIGNAL_HANDLER
+	//the buttons are off-screen, so we sync them up to come down with the shutter
+	animate(src, transform = matrix(), time = SHUTTER_MOVEMENT_DURATION, easing = CUBIC_EASING|EASE_OUT)
+	if(istype(src, /atom/movable/screen/lobby/button/bottom/poll))
+		var/atom/movable/screen/lobby/button/bottom/poll/poll_button = src
+		if(!poll_button.new_poll) //don't activate the poll button unless a poll is up
+			return
+	set_button_status(TRUE)
```

Comment:
ok so like what does this do but also why do we need to think about poll buttons in our expansion proc. typechecks of src like this are a smell that implies an issue with OOP

Reply:
poll buttons start inactive unless a poll is active. it's here so it doesn't get activated when collapsed/expanded

Reply:
it's annoying that the poll button can't make this decision, maybe with a button_displayable proc or something rather then needing to guard against this proc call like you are.

Reply:
we can just not care about letting the players click the buttons while they're being pulled away or brought back into place. not exactly a big issue, to be frank.

Reply:
https://github.com/tgstation/tgstation/pull/76443/commits/7d1dbbcb8e604ae8d176707862cb563c7b32674f